src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpInterop.cs

```diff
@@ -132,46 +141,49 @@ public static async ValueTask CancelationHelper(Task promise, CancellationToken
-                    (Task _promise, JSObject? _fetchResponse) = ((Task, JSObject?))s!;
-                    CancelablePromise.CancelPromise(_promise, static (JSObject? __fetchResponse) =>
+                    (Task _promise, JSObject _jsController) = ((Task, JSObject))s!;
+                    CancelablePromise.CancelPromise(_promise, static (JSObject __jsController) =>
                     {
-                        if (__fetchResponse != null)
+                        if (!__jsController.IsDisposed)
                         {
-                            AbortResponse(__fetchResponse);
+                            AbortResponse(__jsController);
                         }
-                    }, _fetchResponse);
-                }, (promise, fetchResponse)))
+                    }, _jsController);
+                }, (promise, jsController)))
                 {
                     await promise.ConfigureAwait(true);
                 }
             }
             catch (OperationCanceledException oce) when (cancellationToken.IsCancellationRequested)
             {
-                throw CancellationHelper.CreateOperationCanceledException(oce, cancellationToken);
+                throw Http.CancellationHelper.CreateOperationCanceledException(oce, cancellationToken);
             }
             catch (JSException jse)
             {
                 if (jse.Message.StartsWith("AbortError", StringComparison.Ordinal))
                 {
-                    throw CancellationHelper.CreateOperationCanceledException(jse, CancellationToken.None);
+                    throw Http.CancellationHelper.CreateOperationCanceledException(jse, CancellationToken.None);
                 }
                 if (cancellationToken.IsCancellationRequested)
                 {
-                    throw CancellationHelper.CreateOperationCanceledException(jse, cancellationToken);
+                    throw Http.CancellationHelper.CreateOperationCanceledException(jse, cancellationToken);
                 }
                 throw new HttpRequestException(jse.Message, jse);
             }
         }
 
-        public static async ValueTask<T> CancelationHelper<T>(Task<T> promise, CancellationToken cancellationToken, JSObject? fetchResponse = null)
+        public static async Task<T> CancellationHelper<T>(Task<T> promise, CancellationToken cancellationToken, JSObject jsController)
         {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                throw Http.CancellationHelper.CreateOperationCanceledException(null, cancellationToken);
+            }
             if (promise.IsCompletedSuccessfully)
             {
                 return promise.Result;
             }
-            await CancelationHelper((Task)promise, cancellationToken, fetchResponse).ConfigureAwait(true);
-            return await promise.ConfigureAwait(true);
+            await CancellationHelper((Task)promise, cancellationToken, jsController).ConfigureAwait(false);
```

Comment:
nit+unrelated: If we choose a different method name, we don't need to do the cast

Reply:
I like the same name

Reply:
Hmmm, so rather do unnecessary cast, interesting call

Reply:
This is just for compiler to select the right signature, instead of recursion. 
Implicit up-casting would be there anyway in order to re-use the other method. 
And I think it's like NOOP. 

Whatever, I want the code reuse and same name 😛